### PR TITLE
MM-41267: Keywords prompt playbook action

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary

This PR adds a new trigger and new action to channels, refactored from the current Playbooks action: when certain keywords are posted in a channel, prompt the user to run a playbook. The configuration is now per-channel, meaning there is not team-wide keywords triggers as there was before when this lived as a Playbooks action.

Please, review by commit and read the commit message and description, they should be self-explanatory.

For some context: although this is a +874 -979 lines PR, you'll find that it's actually not as bad as it looks: a big bunch of the changes are simply deleting code (https://github.com/mattermost/mattermost-plugin-playbooks/commit/881b198410182cc63eb33b9ef65d2cdae4e169b8 and https://github.com/mattermost/mattermost-plugin-playbooks/commit/e1eeca4cb467904047d165dc1bfb21d55f3456f9) and moving code around (https://github.com/mattermost/mattermost-plugin-playbooks/commit/94e5607772e78d044fc43cc879071b6879a0ef54), so hopefully it's not too annoying to review :)

Note: There's one error in the description of https://github.com/mattermost/mattermost-plugin-playbooks/pull/1081/commits/003bb9b9c5ea7b50e5212339e16d0de24b298d5b: the map is not `TriggerType->(ActionType->Action)` but `TriggerType -> []Action`.

#### Screenshots

| Channel admins (editable) | Regular members (view-only) |
|----|----|
| image | image |

image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41267
https://mattermost.atlassian.net/browse/MM-41268

#### Checklist

- [ ] Telemetry updated: I'll update it as soon as https://github.com/mattermost/mattermost-plugin-playbooks/pull/1078 is merged
- [ ] ~~Gated by experimental feature flag~~
- [X] Unit tests updated

### CODE REVIEW

**Review by crspeller — APPROVED** [review]
Much nicer! Easy to review with all the nice commit descriptions. Perhaps too easy 😆

**Review by iomodo — APPROVED** [review]
Loved it! Great work @agarciamontoro !

File: server/api/actions.go [hunk 1]
```diff
@@ -86,6 +88,52 @@ func (a *ActionsHandler) createChannelAction(w http.ResponseWriter, r *http.Requ
 	ReturnJSON(w, &result, http.StatusCreated)
 }
 
+func isValidTrigger(trigger string) bool {
+	if trigger == "" {
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply server/api/actions.go:92]: This is a helper function to validate the endpoint query parameter, which is retrieved in `parseGetChannelActionsOptions` with `triggerTypeStr := query.Get("trigger_type")`. If that parameters is not present, `triggerTypeStr` will be empty, which means that the caller does not want to filter by trigger.

**Review by abhijit-singh — APPROVED** [review]
Awesome work @agarciamontoro!

Just one non-blocking request and not particularly related to this action, but can we reduce the spacing between the action title & toggle row and the input box to 8px?
image

**Comment by agarciamontoro** [comment]
Thanks, @abhijit-singh! I'll fix that in the PR for the LHS categorization action, as I slightly changed the HTML structure there.

**Pull request merged**